### PR TITLE
Adding image build to workflow

### PR DESCRIPTION
**Pull request by samuelattwood**
- Updated GitHub workflow
- Updated goreleaser config to generate Docker images
- Updated Linux Dockerfile

Signed-off-by: Samuel Attwood <sam@synadia.com>

### CODE REVIEW

**Review by wallyqs — APPROVED** [review]
LGTM

File: docker/linux/Dockerfile [hunk 1]
```diff
@@ -1,15 +1,16 @@
...
-WORKDIR /go/src/prometheus-nats-exporter
-RUN make prometheus-nats-exporter.docker
-
 FROM alpine:latest as osdeps
+
 RUN apk add --no-cache ca-certificates
 
+ARG BINARY=prometheus-nats-exporter
+
 FROM scratch
```
Comment (caleblloyd) [inline docker/linux/Dockerfile:7]: Opinions on using [gcr.io/distroless/static-debian11](https://github.com/GoogleContainerTools/distroless/blob/main/base/README.md) here?  That's essentially what we are making with `scratch`

Comment (wallyqs) [reply docker/linux/Dockerfile:7]: sounds good to me since that includes the certs

Comment (ripienaar) [reply docker/linux/Dockerfile:7]: there's benefit to using same distro - for example @philpennock tracks alpine security updates etc.

Comment (caleblloyd) [reply docker/linux/Dockerfile:7]: I think the only things coming from debian in this image are CA Certificates and TZData.  It has some other niceties, such as a `/tmp` directory and an `/etc/passwd` entry though.

Comment (caleblloyd) [reply docker/linux/Dockerfile:7]: Clearly this particular image has been working fine without those things though, so fine to just leave it the way it is too

**Review by caleblloyd — APPROVED** [review]
LGTM

**Pull request merged**